tests/tests.rs

```diff
@@ -38,40 +38,51 @@ fn test_bool_and() {
 
 #[test]
 fn test_church_mul() {
-  let book = load_lang("church_mul.hvm");
-  let (term, defs, info) = hvm_lang_normal(book, 64);
+  let mut book = load_lang("church_mul.hvm");
+  let (rnet, net, id_map) = hvm_lang_normal(&mut book, 64);
+  let readback = hvm_lang_readback(&net, &book, id_map);
 
-  assert_snapshot!(show_net(&info.net), @"({2 ({2 b c} d) {3 (d e) (e {2 c f})}} (b f))");
-  assert_snapshot!(term.to_string(&defs), @"λa λb (a (a (a (a (a (a b))))))");
-  assert_debug_snapshot!(info.stats.rewrites.total_rewrites(), @"12");
+  assert_snapshot!(show_net(&net), @"({2 ({2 b c} d) {3 (d e) (e {2 c f})}} (b f))");
+  assert_snapshot!(readback, @"λa λb (a (a (a (a (a (a b))))))");
+  assert_debug_snapshot!(rnet.rewrites(), @"12");
 }
 
 #[test]
 fn test_neg_fusion() {
-  let book = load_lang("neg_fusion.hvm");
-  let (term, defs, info) = hvm_lang_normal(book, 512);
+  let mut book = load_lang("neg_fusion.hvm");
+  let (rnet, net, id_map) = hvm_lang_normal(&mut book, 512);
+  let readback = hvm_lang_readback(&net, &book, id_map);
 
-  assert_snapshot!(show_net(&info.net), @"(b (* b))");
-  assert_snapshot!(term.to_string(&defs), @"λa λ* a");
-  assert_debug_snapshot!(info.stats.rewrites.total_rewrites(), @"153");
+  assert_snapshot!(show_net(&net), @"(b (* b))");
+  assert_snapshot!(readback, @"λa λ* a");
+
+  // Todo: investigate why this difference exists
```

Comment:
good handling, comment these kinds of things. just small nitpick, i think the default/standard if `TODO:` with caps